fabric-particles-v1/src/main/java/net/fabricmc/fabric/api/particles/ParticleRegistry.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.particles;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.particles.ParticleRegistryImpl;
+import net.minecraft.client.particle.ParticleFactory;
+import net.minecraft.client.texture.SpriteAtlasTexture;
+import net.minecraft.particle.DefaultParticleType;
+import net.minecraft.particle.ParticleEffect;
+import net.minecraft.particle.ParticleType;
+
+/** Core methods for registering particles with the Fabric API. */
+public interface ParticleRegistry {
+	ParticleRegistry INSTANCE = new ParticleRegistryImpl();
+
+	/** Retrieve the appropriate sprite atlas for particle textures. */
+	SpriteAtlasTexture getParticleSpriteAtlas();
+
+	/** Create a basic particle type that requires no extra information on spawn. */
+	DefaultParticleType createSimpleParticleType();
+
+	/**
+	 * Create a basic particle type that requires no extra information on spawn.
+	 *
+	 * @param shouldAlwaysSpawn Whether this particle should spawn regardless of distance or client settings (a la barrier particles).
+	 */
+	DefaultParticleType createSimpleParticleType(boolean shouldAlwaysSpawn);
```

Comment:
Any reason to expose that it's a DefaultParticleType?

Reply:
Not particularly, just that `ParticleType` is generic and I prefer to avoid returning generic types bare like would be necessary here. I can change it if you prefer.

Reply:
Actually, my bad, there is a reason. Returning a `ParticleType` would not allow the resulting object to be passed to `World#addParticle` without casting, since it expects a `ParticleEffect` (`DefaultParticleType` extends/implements both). We could return a wrapper type of our own, but it's unnecessary.

Reply:
Resolving this conversation for now since the API doesn't work right with this changed, feel free to reopen if you think of a better way.